pkg/controller/che/che_cr_validator.go

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright (c) 2012-2019 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package che
+
+import (
+	"fmt"
+
+	orgv1 "github.com/eclipse/che-operator/pkg/apis/org/v1"
+)
+
+// ValidateCheCR checks Che CR configuration.
+// It should detect:
+// - configurations which miss required field(s) to deploy Che
+// - self-contradictory configurations
+// - configurations with which it is impossible to deploy Che
+func ValidateCheCR(checluster *orgv1.CheCluster, isOpenshift bool) (bool, string) {
+	var isValid bool
+	var errorMessage string
+
+	if !isOpenshift {
+		if checluster.Spec.K8s.IngressDomain == "" {
+			return false, fmt.Sprintf("Required parameter \"Spec.K8s.IngressDomain\" is not set.")
+		}
+	}
+
+	isValid, errorMessage = checkTLSConfiguration(checluster, isOpenshift)
+	if !isValid {
+		return isValid, errorMessage
+	}
+
+	return true, ""
+}
+
+func checkTLSConfiguration(checluster *orgv1.CheCluster, isOpenshift bool) (bool, string) {
+	if !checluster.Spec.Server.TlsSupport {
```

Comment:
Why do we need this check?

Reply:
Because if TLS is turned off no need to check its configuration

Reply:
Then it should be outside function, right?

Reply:
The idea was to have a list of checks:
```
checkTLS
checkABCD
...
```
But I may move that if to the top function.